src/storage.rs

```diff
@@ -1332,44 +1333,51 @@ fn parse_imported_criteria(value: toml::Value) -> Option<CriteriaEntry> {
         .ok()
 }
 
-/// Parse an unparsed audit entry, validating and returning it.
-fn parse_imported_audit(valid_criteria: &[CriteriaName], value: toml::Value) -> Option<AuditEntry> {
-    let mut audit: AuditEntry = parse_from_value(value)
-        .map_err(|err| info!("imported audit parsing failed due to {err}"))
-        .ok()?;
-
+fn retain_only_known_criteria(
+    audit_criteria: &mut Vec<Spanned<CriteriaName>>,
+    valid_criteria: &[CriteriaName],
+) {
     // Remove any unrecognized criteria to avoid later errors caused by being
     // unable to find criteria, and ignore the entry if it names no known
     // criteria.
-    audit
-        .criteria
-        .retain(|criteria_name| is_known_criteria(valid_criteria, criteria_name));
-    if audit.criteria.is_empty() {
+    audit_criteria.retain(|criteria_name| {
+        if !is_known_criteria(valid_criteria, criteria_name) {
+            info!("discarding unknown criteria in imported audit: {criteria_name}");
+            return false;
+        }
+        true
+    });
+
+    if audit_criteria.is_empty() {
         info!("imported audit parsing failed due to no known criteria");
-        return None;
     }
+}
 
-    Some(audit)
+/// Parse an unparsed audit entry, validating and returning it.
+fn parse_imported_audit(valid_criteria: &[CriteriaName], value: toml::Value) -> Option<AuditEntry> {
+    let mut audit_entry = parse_from_value::<AuditEntry>(value).ok()?;
+
+    retain_only_known_criteria(&mut audit_entry.criteria, valid_criteria);
+    if audit_entry.criteria.is_empty() {
+        None
+    } else {
+        Some(audit_entry)
+    }
 }
 
 /// Parse an unparsed wildcard audit entry, validating and returning it.
 fn parse_imported_wildcard_audit(
     valid_criteria: &[CriteriaName],
     value: toml::Value,
 ) -> Option<WildcardEntry> {
-    let mut audit: WildcardEntry = parse_from_value(value)
-        .map_err(|err| info!("imported wildcard audit parsing failed due to {err}"))
-        .ok()?;
+    let mut wildcard_entry = parse_from_value::<WildcardEntry>(value).ok()?;
```

Comment:
This change means we no longer log errors from `parse_from_value` if the parsing failed due to a serde parse error - could we perhaps revert this change to get that log back?

```suggestion
    let mut audit: WildcardEntry = parse_from_value(value)
        .map_err(|err| info!("imported wildcard audit parsing failed due to {err}"))
        .ok()?;
```